sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -903,4 +1048,24 @@ export class KeysClient {
 
     return resultObject;
   }
+
+  private getClonedOptions(
+    options?: ListKeysOptions | RequestOptions | GetKeyOptions | UpdateKeyOptions
+  ): any {
+    if (!options) {
+      return options;
+    }
+
+    return { ...options };
```

Comment:
`...` only does the needful at the top most level. This will not help us we will still be using the `spanOptions` object set by the user

I have updated #4654 such that we use the spread operator at the right level